src/main/java/org/takes/rs/RsWithHeader.java

```diff
@@ -94,22 +95,61 @@ public RsWithHeader(final Response res, final CharSequence name,
      */
     public RsWithHeader(final Response res, final CharSequence header) {
         super(
-            new Response() {
-                @Override
-                public List<String> head() throws IOException {
-                    final List<String> head = new LinkedList<String>();
-                    for (final String hdr : res.head()) {
-                        head.add(hdr);
-                    }
-                    head.add(header.toString());
-                    return head;
-                }
-                @Override
-                public InputStream body() throws IOException {
-                    return res.body();
-                }
-            }
+            new RsWithHeader.ResponseImpl(res, header)
+        );
+    }
+
+    private static final class ResponseImpl implements Response {
+        /**
+         * Pattern for all other lines in the head.
+         */
+        private static final Pattern HEADER = Pattern.compile(
+            "[a-zA-Z\\-]+:\\p{Print}+"
         );
+
+        /**
+         * Original response.
+         */
+        private final transient Response response;
+
+        /**
+         * Header to add.
+         */
+        private final transient CharSequence value;
```

Comment:
@alevohin its value, what we need add to header. In constructor we pass value "header", and I don't want hide local variable with same name. I think about it.